sdk/java-sdk-testkit/src/main/java/kalix/javasdk/testkit/impl/EventSourcedEntityEffectsRunner.java

```diff
@@ -73,6 +74,7 @@ protected <R> EventSourcedResult<R> interpretEffects(
       entity._internalSetEventContext(Optional.of(new TestKitEventSourcedEntityEventContext()));
       for (Object event : EventSourcedResultImpl.eventsOf(effectExecuted)) {
         this._state = handleEvent(this._state, event);
+        entity._internalSetCurrentState(this._state);
```

Comment:
Than, on each applied event, we have a new state that we need to make available to the entity for the next event handler call.